python/amici/sbml_import.py

```diff
@@ -1270,45 +1219,41 @@ def _replace_special_constants(self) -> None:
             (symbol_with_assumptions('avogadro'), sp.sympify(6.02214179e23)),
         ]
         for constant, value in constants:
-            # do not replace if any symbol is shadowing default definition
-            if not any([constant in self.symbols[symbol]['identifier']
-                        for symbol in self.symbols.keys()
-                        if 'identifier' in self.symbols[symbol].keys()]):
-                self._replace_in_all_expressions(constant, value)
-            else:
-                # yes sbml supports this but we wont. Are you really expecting
-                # to be saved if you are trying to shoot yourself in the foot?
-                raise SBMLException(
-                    f'Encountered currently unsupported element id {constant}!'
-                )
+            self._replace_in_all_expressions(constant, value)
```

Comment:
So far it's only one constant, but in general might be faster for `_replace_in_all_expressions` to handle multiple (independent) substitutions at once.

Reply:
True, also applies to other occurrences. Requires rewriting of both `self._replace_in_all_expressions` and `_make_initial`, so let's not do this here.